modules/base/rendering/renderabletrailtrajectory.cpp

```diff
@@ -218,19 +218,19 @@ void RenderableTrailTrajectory::update(const UpdateData& data) {
             // Cap _numberOfVertices in order to prevent overflow and extreme performance
             // degredation/RAM usage
             _numberOfVertices = std::min(
-                static_cast<unsigned int>(timespan / _totalSampleInterval),
+                static_cast<unsigned int>(std::ceil(timespan / _totalSampleInterval)),
                 maxNumberOfVertices
             );
 
             // We need to recalcuate the _totalSampleInterval if _numberOfVertices eqals
             // maxNumberOfVertices. If we don't do this the position for each vertex
             // will not be correct for the number of vertices we are doing along the trail
-            _totalSampleInterval = (_numberOfVertices == maxNumberOfVertices) ?
-                (timespan / _numberOfVertices) : _totalSampleInterval;
+            _totalSampleInterval = std::max(1.0, (_numberOfVertices == maxNumberOfVertices) ?
+                (timespan / _numberOfVertices) : _totalSampleInterval);
```

Comment:
```suggestion
            _totalSampleInterval = (_numberOfVertices == maxNumberOfVertices) ?
                (timespan / _numberOfVertices) : _totalSampleInterval;
            _totalSampleInterval = std::max(1.0, _totalSampleInterval);
```

Suggestion to make it easier to understand